s4e-web/src/app/views/map-view/state/search-results/search-results.service.spec.ts

```diff
@@ -0,0 +1,75 @@
+import {fakeAsync, TestBed, tick} from '@angular/core/testing';
+import {HttpClientTestingModule, HttpTestingController} from '@angular/common/http/testing';
+import {SearchResultsService} from './search-results.service';
+import {SearchResultsStore} from './search-results.store';
+import {SearchResultsQuery} from './search-results.query';
+import {TestingConfigProvider} from '../../../../app.configuration.spec';
+
+describe('SearchResultsService', () => {
+  let searchResultsService: SearchResultsService;
+  let searchResultsStore: SearchResultsStore;
+  let searchResultsQuery: SearchResultsQuery;
+  let http: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      providers: [SearchResultsService, SearchResultsStore, SearchResultsQuery, TestingConfigProvider],
+      imports: [HttpClientTestingModule]
+    });
+
+    http = TestBed.get(HttpTestingController);
+    searchResultsService = TestBed.get(SearchResultsService);
+    searchResultsStore = TestBed.get(SearchResultsStore);
+    searchResultsQuery = TestBed.get(SearchResultsQuery);
+  });
+
+  it('should be created', () => {
+    expect(searchResultsService).toBeDefined();
+  });
+
+  describe('get', () => {
+    it('should correctly handle empty query string', () => {
+      searchResultsService.get('');
+
+      expect(searchResultsQuery.getValue().isOpen).toEqual(false);
+      expect(searchResultsQuery.getValue().queryString).toEqual('');
+
+      http.verify();
+    });
+
+    it('should pass query to endpoint', () => {
+      searchResultsService.get('foo');
+      const req = http.expectOne('api/v1/places?namePrefix=foo');
+      expect(req.request.method).toBe('GET');
+      req.flush({});
+      http.verify();
+    });
+
+    it('should handle non-empty query', fakeAsync(() => {
+      searchResultsService.get('k');
+      const req = http.expectOne('api/v1/places?namePrefix=k');
+      expect(req.request.method).toBe('GET');
+
+      const sampleResult = {
+        latitude: 12,
+        longitude: 23,
+        name: 'kraków',
+        type: 'miasto',
+        voivodeship: 'małopolska'
+      };
+      req.flush({
+        content: [sampleResult]
+      });
+      tick(1000);
+
+      expect(searchResultsQuery.getValue().isOpen).toEqual(true);
+      expect(searchResultsQuery.getValue().queryString).toEqual('k');
+      const ids = searchResultsQuery.getValue().ids;
+      expect(ids.length).toEqual(1);
+      sampleResult['id'] = ids[0]; // ??? ASK MICH
```

Comment:
We talked about it, but I'll repost it here for completeness sake:
I would advice to mock `guid()` via spyOn. I'm not 100% sure it'll work without hitch, as mocking methods from other modules is notoriously problematic. If there is a problem with mocking it you could try extracting call to `guid()` in service to separate service's private method, and mock that one.
It kind of breaks the idea that tests should not affect code structure, but I think it's a reasonable price to pay in this case.